alica_engine/include/engine/constraintmodul/Query.h

```diff
@@ -116,7 +116,7 @@ class Query
 template <class SolverType>
```

Comment:
Should be enough to do a forward declaration for the engine now instead of including it